### PR TITLE
ref: Make getScope private

### PR DESCRIPTION
**Pull request by HazAT**

### CODE REVIEW

File: tests/State/HubTest.php [hunk 1]
```diff
@@ -292,4 +292,13 @@ public function testCaptureEvent(): void
 
         $this->assertEquals('2b867534eead412cbdb882fd5d441690', $hub->captureEvent(['message' => 'test']));
     }
+
+    private static function getScope($hub): Scope
```
Comment (Jean85) [inline tests/State/HubTest.php]: Why `static`?

Also:
```suggestion
    private static function getScope(Hub $hub): Scope
```

Comment (ste93cry) [reply tests/State/HubTest.php]: Even better, `HubInterface`

File: tests/State/HubTest.php [hunk 2]
```diff
@@ -292,4 +293,13 @@ public function testCaptureEvent(): void
 
         $this->assertEquals('2b867534eead412cbdb882fd5d441690', $hub->captureEvent(['message' => 'test']));
     }
+
+    protected function getScope(HubInterface $hub): Scope
```
Comment (ste93cry) [inline tests/State/HubTest.php]: Since this class is `final` it makes no sense to have protected methods

File: tests/State/HubTest.php [hunk 3]
```diff
@@ -292,4 +293,13 @@ public function testCaptureEvent(): void
...
+    {
+        $class = new \ReflectionClass($hub);
```
Comment (ste93cry) [inline tests/State/HubTest.php]: We can use `ReflectionMethod` directly instead of passing for the class metadata

File: tests/SdkTest.php [hunk 4]
```diff
@@ -94,7 +94,13 @@ public function testAddBreadcrumb(): void
 
         addBreadcrumb($breadcrumb);
 
-        $this->assertSame([$breadcrumb], Hub::getCurrent()->getScope()->getBreadcrumbs());
+        $hub = Hub::getCurrent();
+
+        $class = new \ReflectionClass($hub);
```
Comment (ste93cry) [inline tests/SdkTest.php]: Same as above, we can use `ReflectionMethod` directly

File: tests/ClientTest.php [hunk 5]
```diff
@@ -274,8 +274,14 @@ public function testHandlingExceptionThrowingAnException(): void
 
         $client = new Client(new Options(), $transport, $this->createEventFactory());
 
-        Hub::getCurrent()->bindClient($client);
-        $client->captureException($this->createCarelessExceptionWithStacktrace(), Hub::getCurrent()->getScope());
+        $hub = Hub::getCurrent();
+        $hub->bindClient($client);
+
+        $class = new \ReflectionClass($hub);
```
Comment (ste93cry) [inline tests/ClientTest.php]: We can use `ReflectionMethod` directly

**Comment by ste93cry** [comment]
I don't really like that we have to use reflection to test the internal behavior of an object, however at the moment I don't see any other way around so LGTM

**Pull request merged**